inc/Config/QueryContext/HttpQueryContext.php

```diff
@@ -89,15 +150,16 @@ public function get_datasource(): HttpDatasource {
 	 * Override this method to specify a custom endpoint for this query.
 	 */
 	public function get_endpoint( array $input_variables ): string {
-		return $this->get_datasource()->get_endpoint();
+		$endpoint = $this->config['endpoint'] ?? $this->get_datasource()->get_endpoint();
```

Comment:
not a big fan of this logic gate. there should be a singular source

Reply:
I certainly agree. I see all of this as an interim state caught between our original understanding, "all code", and the growth of the persisted pathes.

I'm just trying to keep it all floating for next week right now.

Reply:
Can we just exempt datasource properties from config for now? It would keep the surface area smaller and help us move a bit more carefully

Reply:
I'm not sure I understand what you mean. I needed this one, `endpoint`, specifically when attempting to transform what we have in the DB into something that executes.

Reply:
We can create an `AirtableGetItemQuery` (keeping the static config changes in this PR, they're great) to implement endpoint and image_url. I don't think the approach to endpoint and image_url in this PR is a shippable solution.